src/index.ts

```diff
@@ -423,8 +432,29 @@ export class UnleashClient extends TinyEmitter {
         await this.storage.save(storeKey, toggles);
     }
 
+    private isUpToDate() {
+        if (this.togglesStorageTTL <= 0 || this.toggles.length === 0) {
+            return false;
+        }
+        const timestamp = Date.now();
+
+        return !!(
+            this.lastRefreshTimestamp &&
+            timestamp - this.lastRefreshTimestamp <= this.togglesStorageTTL
+        );
+    }
+
+    private async updateLastRefresh() {
+        this.lastRefreshTimestamp = Date.now();
+        this.storage.save(lastUpdateKey, this.lastRefreshTimestamp);
+    }
+
     private async fetchToggles() {
         if (this.fetch) {
+            if (this.isUpToDate()) {
+                this.emitReady();
+                return;
```

Comment:
This method have been carefully designed to not use "early returns". Early return in combination with try/catch control flow increases the complexity. I am not comfortable about breaking the current pattern to support avoiding the initial fetch use case.

I think we instead can have a specific `initalFetchMethod`, that only calls `fetchToggles` if the cached data is not considered recent.

Reply:
By running this on each fetch we have another potential bug in the future: What if the stored timestamp somehow ends up far in to the future (we are dealing with untrusted browsers) we will never try to fetch updates at all.

Reply:
Yes you're right, the `initialFetchMethod` is a better idea than this, we will rework it.

Come to think of it, this code is actually buggy... because if we update the context and the `isUpToDate` function says "yes", the flags will not be updated, which is not intended.

Reply:
Anything in user-land cannot be trusted. Users might suddenly change their local time etc, which forces us to be a bit restrictive on what we trust. E.g. if the stored data is 2 hours in the future we should probably disregard the cached data and do a fetch.